### PR TITLE
Fixed #31382 -- Made Model.save(update_fields=...) raise ValueError on non-concrete fields.

### PR DESCRIPTION
**Pull request by patgarcia**
https://code.djangoproject.com/ticket/31382

### CODE REVIEW

File: tests/update_only_fields/fields.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    def contribute_to_class(self, cls, name, **kwargs):
+        super().contribute_to_class(cls, name, **kwargs)
+        setattr(cls, name, self)
+
+    def get_attname_column(self):
+        attname, _ = super().get_attname_column()
+        return attname, None
+
+    def get_default(self):
+        return ("", "")
```
Comment (felixxm) [inline tests/update_only_fields/fields.py:43]: We don't need a full implementation of a compound field for this test, I would add a simple non-concrete field directly in models, e.g.

```python
class NonConcreteField(models.IntegerField):
    def db_type(self, connection):
        return None

    def get_attname_column(self):
        attname, _ = super().get_attname_column()
        return attname, None

class Profile(models.Model):
    ...
    non_concrete = NonConcreteField()
```

File: django/db/models/base.py [hunk 2]
```diff
@@ -735,7 +735,7 @@ def save(self, force_insert=False, force_update=False, using=None,
 
             if non_model_fields:
                 raise ValueError("The following fields do not exist in this "
-                                 "model or are m2m fields: %s"
+                                 "model or are m2m fields, or non-concrete: %s"
```
Comment (felixxm) [inline django/db/models/base.py:738]:
```suggestion
                                 "model or are m2m or non-concrete fields: %s"
```

**Comment by M-050** [comment]
Ml2811911@gmail.com

**Comment by felixxm** [comment]
@patgarcia Thanks :+1: Welcome aboard :boat:

**Pull request merged**